### PR TITLE
load_mon: increase cpuload update rate, get corresponding time safetly, and misc cleanup

### PR DESCRIPTION
**Pull request by dagar**
- increase rate from 1 Hz to 3.3 Hz (every 300 ms)
 - cpu load calculation grab timestamp with runtime in a critical section
 - task stack only check one task per cycle (but cycle at a higher rate)
 - decrease available FD threshold 3 -> 2
 - minor cleanup (split out header)

### CODE REVIEW

**Comment by dagar** [comment]
I increased the rate slightly because I wanted to have a bit more detail in the logs.

File: src/modules/load_mon/LoadMon.cpp [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+		_task_stack_info_pub.publish(task_stack_info);
+
+		// Found task low on stack, report and exit. Continue here in next cycle.
+		if (stack_free < STACK_LOW_WARNING_THRESHOLD) {
+			PX4_WARN("%s low on stack! (%i bytes left)", task_stack_info.task_name, stack_free);
+		}
+
+		// Found task low on file descriptors, report and exit. Continue here in next cycle.
+		if (fds_free < FDS_LOW_WARNING_THRESHOLD) {
+			PX4_WARN("%s low on FDs! (%i FDs left)", task_stack_info.task_name, fds_free);
```
Comment (julianoes) [inline src/modules/load_mon/LoadMon.cpp:192]: These warnings will now appear 3 times per second and spam the log quite a bit, right?

File: src/modules/load_mon/LoadMon.cpp [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+int LoadMon::print_usage(const char *reason)
+{
+	if (reason) {
+		PX4_ERR("%s\n", reason);
+	}
+
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+Background process running periodically with 1 Hz on the low priority work queue to calculate the CPU load and RAM
```
Comment (julianoes) [inline src/modules/load_mon/LoadMon.cpp:209]: The 1 Hz is now a lie, right?

File: src/modules/load_mon/LoadMon.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/load_mon/LoadMon.cpp:192]: > These warnings will now appear 3 times per second and spam the log quite a bit, right?

Only in the worst case if running the max number of tasks (32 or 64) and every one of them was either low on stack or FDs.

Overall it's kind of a lateral move. Currently in master it cycles at 1 Hz, but each iteration checks 2 tasks, and if a task entry (an array of CONFIG_MAX_TASKS) is empty it jumps forward. Now it runs slightly faster for the sake of cpuload measurement, but it's doing less work per iteration checking tasks.

File: src/modules/load_mon/LoadMon.cpp (hunk 2 above)
Comment (dagar) [reply src/modules/load_mon/LoadMon.cpp:209]: It is!

Comment (dagar) [reply src/modules/load_mon/LoadMon.cpp]: Removed.

**Pull request merged**